### PR TITLE
[KeyVault]Change encoding way from ASCII/Default to UTF8

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Migration guide:

For encrypted/wrapped result which is operated in Az.KeyVault<5.0.0, to correct decrypted/unwrapped result, please use
Previous:
```
$result = Invoke-AzKeyVaultKeyOperation -Operation Decrypt -Algorithm RSA1_5 -VaultName test-kv -Name test-key -Value (ConvertTo-SecureString -String $encryptedResult. -AsPlainText -Force)
$plainText = $result.Result
```
Now: 
```
$result = Invoke-AzKeyVaultKeyOperation -Operation Decrypt -Algorithm RSA1_5 -VaultName test-kv -Name test-key -Value (ConvertTo-SecureString -String $encryptedResult -AsPlainText -Force)
$plainText = [System.Text.Encoding]::ASCII.GetString($result.RawResult) 
```

File: src/KeyVault/KeyVault/Models/Key/PSKeyOperationResult.cs [hunk 1]
```diff
@@ -13,34 +13,46 @@ public class PSKeyOperationResult
         // Summary: encryted result or wraped result is base64 format. decryted result or unwraped result is plain text
         public string Result { get; }
 
+        /// <summary>
+        /// For encrypt operation, this is the byte array format for the result of the encryption.
+        /// For decrypt operation, this is the byte array format for the decrypted data.
+        /// For wrap operation, this is the byte array of wrapped key.
+        /// For unwrap operation, this is the byte array of unwrapped key.
+        /// </summary>
+        public byte[] RawResult { get; }
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/Models/Key/PSKeyOperationResult.cs:22]: Why do we expose RawResult to public? It'll be difficult if we want to change it in the future.

Comment (BethanyZhou) [reply src/KeyVault/KeyVault/Models/Key/PSKeyOperationResult.cs]: Make sense.

**Review by isra-fel — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Comment by BethanyZhou** [comment]
Seems it's a random failure in https://github.com/Azure/azure-powershell/blob/3504d69b9371bf0a79662ef58ec5d70b675f62b3/src/Sql/Sql.Test/ScenarioTests/DistributedAvailabilityGroupTests.ps1#L37

**Pull request merged**